### PR TITLE
PWA-707: Update Venia's accordions to match the style guides accordions

### PR DESCRIPTION
**Pull request by davemacaulay**
## Description

Update accordions throughout the application to match those of the style guide.

## Related Issue
https://jira.corp.magento.com/browse/PWA-707

### Verification Stakeholders
@jimbo 
@dpatil-magento 
@anthoula 

### Verification Steps
1. Navigate to cart and checkout and verify the accordions meet the style guide
2. Verify all instances of the style guide are working as expected
3. Verify instances of black links have been updated to blue links (such as Sign in on cart)

## Checklist

- [x] I have added tests to cover my changes, if necessary.
- [x] I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2527.pwa-venia.com : LH Performance Expected 0.85 Actual 0.51, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 83.333333333333
https://pr-2527.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2527.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.41, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 50

File: packages/venia-ui/lib/components/Accordion/section.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const Section = props => {
 
     const isOpen = openSectionIds.has(id);
     const titleIconSrc = isOpen ? ArrowUp : ArrowDown;
-    const titleIcon = <Icon src={titleIconSrc} />;
+    const titleIcon = <Icon src={titleIconSrc} size={24} />;
```
Comment (sirugh) [inline packages/venia-ui/lib/components/Accordion/section.js:22]: Looks like the default is 24 anyways, at least per the snaps. I do like being explicit though 👍

**Comment by davemacaulay** [comment]
> Code looks good, the only thing I noticed was that the "Search" text turns blue when the search is active/open:
> 
> Screen Shot 2020-07-08 at 10 00 20 AM
> 
> Was that intentional?

@supernova-at this was actually introduced in another change and is present on develop: https://develop.pwa-venia.com

**Comment by davemacaulay** [comment]
@sirugh we're using the browsers default built in `:focus` styles (as per @jimbo's request). Modifying these styles results in us having to declare a lot of things about the focus state, we're also unable to give the focus state a border radius in most browsers. Due to this I'd say the small placement issue is fine, unless you know a solution to fix this without having to specify the outline.

We have a separate story for all things card related 😄  https://jira.corp.magento.com/browse/PWA-706

**Comment by sirugh** [comment]
> We have a separate story for all things card related 😄 https://jira.corp.magento.com/browse/PWA-706

Dos the items review component count as an accordion? It's kind of a mix of both card and accordion, if you have more than a few items in your cart.

**Comment by dpatil-magento** [comment]
@davemacaulay `Sign-out` in left drawer, `Remove` in cart accordions, `Edit` in checkout cards (shipping) - these links should be changed to blue?   rest all looks good.

**Comment by davemacaulay** [comment]
> @davemacaulay `Sign-out` in left drawer, `Remove` in cart accordions, `Edit` in checkout cards (shipping) - these links should be changed to blue? rest all looks good.

@dpatil-magento sign out should remain black, as it's a different type of link.
I've updated the Remove links to use the new blue links.
Edit in checkout cards will be handled by the Cards story which Ievgen is working on 😄

**Pull request merged**